### ISSUE TITLE
[Server] Macro of Zabbix can't be replaced.

### ISSUE THREAD

**Issue by bynnchapu** [project-hatohol/hatohol#326]
An user says the user want that Macro of Zabbix is replaced.
Example:
```
When name of the host is "Zabbix Server":
Before: Too many processes on {HOST.NAME}
After: Too many processes on "Zabbix Server".
```

**Comment by cosmo0920** [comment]
:heavy_check_mark: For now, macro of Zabbix can be replaced with expanded information which are gotten by ZabbixAPI.
:x: For now, macro of Zabbix CAN NOT be replaced with expanded information which are gotten by Hap version of ZabbixAPI yet.

**Comment by cosmo0920** [comment]
Above comments are present state about `triggers_view`.

**Comment by cosmo0920** [comment]
Now, `triggers_view` can expand Zabbix macro.

**Comment by kz0817** [comment]
> Now, triggers_view can expand Zabbix macro.
I haven't seen the screen yet though... Can it also show unexpanded macro ?

**Comment by cosmo0920** [comment]
>> Now, triggers_view can expand Zabbix macro.

> I haven't seen the screen yet though... Can it also show an unexpanded macro ?

Ah..., expand macro information is gotten at same time to get `trigger_info`.
Therefore, macros which are gotten before these fixes into server are not to be expanded.

**Comment by cosmo0920** [comment]
> Can it also show an unexpanded macro ?

Yes, it can. 
When hatohol does not have expanded macro information, it shows unexpanded macro.

**Comment by kz0817** [comment]
Thank you for reply.
Can we see an unexpanded macro even if hatohol has the expanded macro ?

**Comment by cosmo0920** [comment]
> Can we see an unexpanded macro even if hatohol has the expanded macro ?

Sadly, it cannot.
Either unexpanded macro or expanded macro shows in hatohol Web UI.

**Comment by kz0817** [comment]
> Now, triggers_view can expand Zabbix macro.

Thank you. I see the behavior.

**Comment by cosmo0920** [comment]
:heavy_check_mark:  For now, macro of Zabbix can be replaced with expanded information which are gotten by Hap version of ZabbixAPI.

**Comment by cosmo0920** [comment]
`trigger` and `event` pages shows expanded descriptions if avaliable.
It seems that Zabbix API does not have macro in `item`. 
I have no idea to treat Zabbix macro in `item` page.

**Comment by cosmo0920** [comment]
Oh, I've noticed that there is macro within posted ticket in Redmine....

![screenshot from 2015-03-04 10 25 24](https://cloud.githubusercontent.com/assets/700876/6476427/e4df1fe8-c258-11e4-9ec4-27bae4b85f5a.png)

**Comment by ashie** [comment]
> And I checked the response of the example, I think it is good we should store value of "key_", and use it.
> Example of API response (from Zabbix API document)

Hatohol has an implementation to expand it, but there was a bug in it.
I fixed it in #1153.
